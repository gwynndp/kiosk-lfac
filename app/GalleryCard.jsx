import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledTitle = styled('h1')`
  display: flex;
  align-items: flex-end;
  font-size: 18px;
  text-transform: uppercase;
  height: 40px;
  margin: 5px;
  border-bottom: 1px solid black;
`;

const Image = styled('img')`
  width: 190px;
  height: 250px;
`;

const Card = styled.div`
  width: 190px;
  background-color: white;
  padding: 10px;
  margin: 20px;
`;

export default function GalleryCard({ video, handleVideoCardClick }) {
  return (
    <Card
      onClick={() => {
        handleVideoCardClick(video);
      }}
    >
      <StyledTitle>{video.name}</StyledTitle>
      <Image src={video.img} alt={`Video titled ${video.name}`} />
    </Card>
  );
}

GalleryCard.propTypes = {
  video: PropTypes.objectOf(
    PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.func])
  ),
  handleVideoCardClick: PropTypes.func.isRequired,
};
