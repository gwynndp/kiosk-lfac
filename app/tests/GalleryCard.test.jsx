import React from 'react';
import renderer from 'react-test-renderer';
import GalleryCard from '../GalleryCard';

test('should render a GalleryCard', () => {
  const props = {
    key: 4,
    name: 'Ashley Sanchez',
    img: 'images/videos/Ashley-Sanchez.jpg',
    video: {
      id: 4,
      category: 'people',
      name: 'Ashley Sanchez',
      src: 'videos/Ashley-Sanchez.mp4',
      img: 'images/videos/Ashley-Sanchez.jpg'
    },
    handleVideoCardClick: () => {}
  };
  const tree = renderer
    .create(
      <GalleryCard
        key={props.id}
        name={props.name}
        img={props.img}
        video={props.video}
        handleVideoCardClick={props.handleVideoCardClick}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

// should direct to a VideoPage when clicked  //DOES THIS TEST GO HERE?
