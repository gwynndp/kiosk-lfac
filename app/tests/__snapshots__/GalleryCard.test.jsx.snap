// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a GalleryCard 1`] = `
<div
  className="sc-AykKE jDVYoJ"
  onClick={[Function]}
>
  <h1
    className="sc-AykKC flIKka"
  >
    Ashley Sanchez
  </h1>
  <img
    alt="Video titled Ashley Sanchez"
    className="sc-AykKD kcIegN"
    src="images/videos/Ashley-Sanchez.jpg"
  />
</div>
`;
